frontend/src/components/widgets/DataFrame/columns/DateColumn.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GridCell } from "@glideapps/glide-data-grid"
+
+import {
+  BaseColumn,
+  BaseColumnProps,
+  getDateCell,
+  getDateCellContent,
+} from "src/components/widgets/DataFrame/columns/utils"
+import {
+  DatetimePickerCell,
+  PythonDateType,
+} from "src/components/widgets/DataFrame/customCells/DatetimePickerCell"
+
+export interface DateColumnParams {
+  readonly format?: string
+}
+
+function DateColumn(props: BaseColumnProps): BaseColumn {
+  return {
+    ...props,
+    kind: "date",
+    sortMode: "smart",
```

Comment:
I think we can use `default` here since the ISO8605 string can just be sorted as string.